org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/apt/model/ElementsImpl9.java

```diff
@@ -284,6 +288,37 @@ private JavaFileObject getSourceJavaFileObject(ReferenceContext referenceContext
 		return null;
 	}
 	@Override
+	public TypeElement getEnumConstantBody(VariableElement enumConstant) {
+		switch(enumConstant.getKind()) {
+			case ENUM_CONSTANT -> {
+				VariableElementImpl variableImpl = (VariableElementImpl) enumConstant;
+				if (variableImpl._binding instanceof FieldBinding field) {
+					FieldDeclaration sourceField = field.sourceField();
+					if (sourceField == null)
+						return null;
+					if (sourceField.initialization instanceof QualifiedAllocationExpression qualAlloc) {
+						TypeDeclaration decl = qualAlloc.anonymousType;
+						if (decl == null)
+							return null;
+
+						SourceTypeBinding declaringClass = (SourceTypeBinding) field.declaringClass;
+						TypeDeclaration enumTypeDecl = declaringClass.scope.referenceContext;
+						// The following code is copied from BlockScope#addAnonymousType() to force
+						// resolution of the local type binding ahead of a full CU#resolve()
+						BlockScope scope = field.isStatic() ? enumTypeDecl.staticInitializerScope : enumTypeDecl.initializerScope;
+						if (scope != null) {
+							ClassScope anonymousClassScope = new ClassScope(scope, decl);
+							anonymousClassScope.buildAnonymousTypeBinding(scope.enclosingSourceType(), declaringClass);
```

Comment:
`scope.enclosingSourceType()` looks a bit like going in a circle, like: isn't this the same as `declaringClass`?

Reply:
It is. And admit, it was lazy coding although I did observe they are same :)